src/System.Linq.Expressions/src/System/Dynamic/CallInfo.cs

```diff
@@ -86,7 +87,7 @@ public override int GetHashCode()
         public override bool Equals(object obj)
         {
             var other = obj as CallInfo;
-            return other != null && ArgumentCount == other.ArgumentCount && ArgumentNames.ListEquals(other.ArgumentNames);
+            return other != null && ArgumentCount == other.ArgumentCount && ArgumentNames.SequenceEqual(other.ArgumentNames);
```

Comment:
This could hurt performance.  LINQ's SequenceEqual's will compare all of the elements by getting an enumerator for each and MoveNext/Current'ing along both.  In contrast, the ListEquals method you've deleted works in terms of ReadOnlyCollection, avoids interface calls, avoids those enumerator allocations, and indexes directly into the collection rather than enumerating.  I suggest this be reverted unless there's something I'm missing.